src/ride_gui.cpp

```diff
@@ -244,6 +241,7 @@ void RideSelectGui::OnClick(WidgetNumber wid_num, const Point16 &pos)
 					_rides_manager.NewInstanceAdded(instance);
 					ShowCoasterManagementGui(ri);
 				}
+				delete this;
```

Comment:
This has nothing to do with fixing the assert as far as I can see, so it doesn't quite belong in this commit.

Is there a rationale why removing the selection window after making a selection improves UI?

Reply:
It slipped in by accident, though I'd still like to propose this change when it's already there. I believe most of the time, people build one ride, then take their time to set up paths and queues, change the ride's pricing, colouring and settings, etc. For tracked rides, they additionally spend a lot of time designing the track and running the ride in test mode first. Most of this is not implemented yet, but the Ride Selection window is not used again for quite some time after settling on which ride to construct, and it only blocks the view on the map unnecessarily. So I would prefer that this window auto-closes after the user hits Select. I don't feel strongly about this though.

Reply:
The point is, if you read the list commit messages, it doesn't say "auto-close ride selection window" ever, so finding the commit that changes the behavior becomes a search party, probably with bi-section of by searching in git for changes on that line.

I have no string feeling about either, so if you want it in, that's fine. Please make it a separate commit, to enhance chances of finding it again if the need arises.

Reply:
moved to #159